lib/index.js

```diff
@@ -47,8 +49,190 @@ api.validateConfiguration = (validatorConfig, callback) =>
   validate('ledger-validator-multiproof', validatorConfig, callback);
 
 api.validateEvent = (event, validatorConfig, callback) => {
-  // determine which proof types should be checked
-
-  // FIXME: Implement
-  callback();
+  async.auto({
+    schema: callback => callback(),
+    signatures: ['schema', (results, callback) =>
+      _validateSignatures(event, callback)]
+  }, err => callback(err));
 };
+
+function _validateSignatures(event, callback) {
+
+  // FIXME: could these parameters just be in the ledger configuration instead
+  // of inventing `ParameterAlgorithm`?
```

Comment:
No, because we want them to be dynamic. They should change over time without having to agree on a new configuration for the ledger.

Reply:
@dlongley can you say more about this?  Who is 'we' in this case, and who is in control of the ledger configuration?  And wouldn't it be a good thing if it were an agreed upon change but a group of actors?

Reply:
What I mean is that the parameters should be arrived at algorithmically. That algorithm should be decided upon by consensus of course, but it should be an algorithm instead of a set of constants, nonetheless.

Reply:
put veresone p/k requirement in config.

Reply:
done.